edition = "2023";

import "common.proto";

package aim;

message CrosshairTheme {
  StoredColor color = 1;
  StoredColor outline_color = 2;
}

message WallTexture {
  string texture_name = 1;
  float scale = 2;
}

message WallAppearance {
  oneof type {
    StoredColor color = 1;
    WallTexture texture = 2;
  }
  StoredColor mix_color = 3;
  // Defaults to 0. The amount of the color to mix in. Similar to lerp or mix in shaders.
  float mix_percent = 4;
}

message HealthBarAppearance {
  StoredColor health_color = 1;
  float health_alpha = 2;

  StoredColor background_color = 3;
  float background_alpha = 4;
}

message Theme {
  string name = 1;

  // This theme is a reference to another theme. No other fields should be set when this value is
  // used. Useful for defining a DefaultStatic theme that can be changed to point to different
  // themes whenever you want.
  string reference = 10;

  WallAppearance roof_appearance = 2;
  WallAppearance side_appearance = 3;
  WallAppearance front_appearance = 4;
  WallAppearance floor_appearance = 5;
  WallAppearance back_appearance = 6;

  CrosshairTheme crosshair = 7;
  StoredColor target_color = 8;
  StoredColor ghost_target_color = 9;

  HealthBarAppearance health_bar = 11;
}
